packages/block-library/src/template-part/edit/placeholder/index.js

```diff
@@ -35,6 +40,17 @@ export default function TemplatePartPlaceholder( { setAttributes } ) {
 		} );
 	}, [ setAttributes ] );
 
+	// If there are inner blocks already present, this is a convert to template part flow.
+	// Therefore immediately create the template part with the given inner blocks.
+	useEffect( () => {
+		if ( innerBlocks.length ) {
+			onCreate();
+		}
+	}, [] );
+	if ( innerBlocks.length ) {
+		return <Spinner />;
+	}
```

Comment:
Since the placeholder has updated a lot since this PR was originally made, I had to make some changes for this transform to work properly.  Here, we pass down the innerBlocks to the placeholder and immediately create the entity if they exist.  The idea is that this block will never render the placeholder when innerBlocks are present, unless we are in the conversion flow.